tests/vmi_multus_test.go

```diff
@@ -1010,59 +1015,98 @@ func cidrToIP(cidr string) string {
+func configInterface(vmi *v1.VirtualMachineInstance, interfaceName, interfaceAddress string, userModifierPrefix ...string) error {
+	if vmi == nil {
+		return fmt.Errorf("the VMI must exist for us to configure an interface in it")
+	}
+
 	setStaticIpCmd := fmt.Sprintf("%sip addr add %s dev %s\n", strings.Join(userModifierPrefix, " "), interfaceAddress, interfaceName)
-	err := console.SafeExpectBatch(vmi, []expect.Batcher{
-		&expect.BSnd{S: "\n"},
-		&expect.BExp{R: console.PromptExpression},
-		&expect.BSnd{S: setStaticIpCmd},
-		&expect.BExp{R: console.PromptExpression},
-		&expect.BSnd{S: "echo $?\n"},
-		&expect.BExp{R: console.RetValue("0")},
-	}, 15)
-	Expect(err).ToNot(HaveOccurred(), "Failed to configure address %s for interface %s on VMI %s", interfaceAddress, interfaceName, vmi.Name)
+	if err := runSafeCommand(vmi, setStaticIpCmd); err != nil {
+		return fmt.Errorf("Failed to configure address %s for interface %s on VMI %s: %w", interfaceAddress, interfaceName, vmi.Name, err)
+	}
 
-	setIfaceUpCmd := fmt.Sprintf("ip link set %s up\n", interfaceName)
-	err = console.SafeExpectBatch(vmi, []expect.Batcher{
-		&expect.BSnd{S: "\n"},
-		&expect.BExp{R: console.PromptExpression},
-		&expect.BSnd{S: setIfaceUpCmd},
-		&expect.BExp{R: console.PromptExpression},
-		&expect.BSnd{S: "echo $?\n"},
-		&expect.BExp{R: console.RetValue("0")},
-	}, 15)
-	Expect(err).ToNot(HaveOccurred(), "Failed to set interface %s up on VMI %s", interfaceName, vmi.Name)
+	return setUpInterface(vmi, interfaceName)
+}
+
+func configSRIOVInterface(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance, interfaceMac, interfaceAddress string) error {
+	if vmi == nil {
+		return fmt.Errorf("the VMI must exist for us to configure an interface in it")
+	}
+
+	err, interfaceName := getSRIOVInterfaceNameByMAC(virtClient, vmi, interfaceMac)
+	if err != nil {
+		return fmt.Errorf("should get sriov interface: %w", err)
+	}
+
+	return configInterface(vmi, interfaceName, interfaceAddress)
+}
+
+func checkInterface(vmi *v1.VirtualMachineInstance, interfaceName string) error {
+	if vmi == nil {
+		return fmt.Errorf("the VMI must exist for us to check an interface in it")
+	}
+
+	cmdCheck := fmt.Sprintf("ip link show %s\n", interfaceName)
+	if err := runSafeCommand(vmi, cmdCheck); err != nil {
+		return fmt.Errorf("Interface %s was not found in the VMI %s within the given timeout: %w", interfaceName, vmi.Name, err)
+	}
+
+	return nil
 }
 
-func checkInterface(vmi *v1.VirtualMachineInstance, interfaceName string) {
+func checkMacAddress(vmi *v1.VirtualMachineInstance, interfaceName, macAddress string) error {
 	cmdCheck := fmt.Sprintf("ip link show %s\n", interfaceName)
-	err := console.SafeExpectBatch(vmi, []expect.Batcher{
+	if err := console.SafeExpectBatch(vmi, []expect.Batcher{
 		&expect.BSnd{S: "\n"},
 		&expect.BExp{R: console.PromptExpression},
 		&expect.BSnd{S: cmdCheck},
-		&expect.BExp{R: console.PromptExpression},
+		&expect.BExp{R: macAddress},
```

Comment:
dont you need RetValue(mac...) ?
so it wont consume itself

Reply:
that one I only changed to return an error, I didn't change the content of the command. I think github diff shows that is a mistake. please take a look at the code before the change: here:https://github.com/kubevirt/kubevirt/blob/master/tests/vmi_multus_test.go#L1055

I agree me might need retValue here, but I prefer in a different PR. Is that acceptable?

Reply:
It does not need a RetValue, please learn how the safe expecter work.
Using the RetValue will require a messy regex here.

If you do have an explicit issue you identify, please share it.